kernel-rpm-script

```diff
@@ -0,0 +1,302 @@
+		$copy_or_link .."$modulesdir"/$x /boot/$x-"$kernelrelease"-"$flavor" || script_rc=$?
+		if [ -e "$modulesdir"/.$x.hmac ]; then
+		    $copy_or_link .."$modulesdir"/.$x.hmac /boot/.$x-"$kernelrelease"-"$flavor".hmac || script_rc=$?
+		fi
+	    fi
+	done
+    fi
+
+    # Add symlinks of compatible modules to /lib/modules/$krel/weak-updates/,
+    # run depmod and mkinitrd
+    if [ -x $wm2 ]; then
+	if [ "$basepackage" = 1 ]; then
+	    run_wm2 --add-kernel "$kernelrelease"-"$flavor" || script_rc=$?
+	else
+	    rpm -ql $nvr | run_wm2 --add-kernel-modules "$kernelrelease"-"$flavor" || script_rc=$?
+	fi
+    else
+	echo "$wm2 does not exist, please run depmod and mkinitrd manually" >&2
+	script_rc=1
+    fi
+
+    if [ ! -e /.buildenv ] ; then
+	if [ -f /etc/fstab ] ; then
+	    # only run the bootloader if the usual bootloader configuration
+	    # files are there -- this is different on every architecture
+	    bl_rc=-1
+	    initrd=initrd-"$kernelrelease"-"$flavor"
+	    if [ "$flavor" = rt ]; then
+		default=force-default
+	    fi
+	    if [ -e /boot/$initrd -o ! -e "$modulesdir" ] && run_bootloader ; then
+		[ -e /boot/$initrd ] || initrd=
+		if [ -x /usr/lib/bootloader/bootloader_entry ]; then
+		    /usr/lib/bootloader/bootloader_entry \
+			add \
+			"$flavor" \
+			"$kernelrelease"-"$flavor" \
+			"$image"-"$kernelrelease"-"$flavor" \
+			$initrd \
+			$default || script_rc=$?
+		else
+		    message_install_bl
+		fi
+	    fi
+	else
+	    message_install_bl
+	fi
+    fi
+
+    [ -z "$certs" ] || [ "$basepackage" -ne 1 ] || /usr/lib/suse-module-tools/kernel-cert-$op --certs "$certs" "$@" || script_rc=$?
+
+    ;;
+postun)
+    # If a kernel package is removed before the next reboot, we assume that the
+    # multiversion variable in /etc/zypp/zypp.conf is not configured and we delete
+    # the flag again (fate#312018)
+    rm -f /boot/do_purge_kernels
+
+    if [ -e "$systemmap" ]; then
+	# the same package was reinstalled or just rebuilt, otherwise the files
+	# would have been deleted by now
+	# do not remove anything in this case (bnc#533766)
+	rm -f /var/run/rpm-$nvr-modules
+	exit 0
+    fi
+
+    if [ "$basepackage" = 0 ]; then
+	if [ -x $wm2 ]; then
+	    run_wm2 --remove-kernel-modules "$kernelrelease"-"$flavor" < /var/run/rpm-$nvr-modules
+	fi
+	rm -f /var/run/rpm-$nvr-modules
+	exit 0
+    fi
+    # Remove symlinks from "$modulesdir"/weak-updates/.
+    if [ -x $wm2 ]; then
+	run_wm2 --remove-kernel "$kernelrelease"-"$flavor"
+    fi
+
+    # remove fstab check once perl-Bootloader can cope with it
+    if [ -f /etc/fstab ]; then
+	if [ -x /usr/lib/bootloader/bootloader_entry ]; then
+	    /usr/lib/bootloader/bootloader_entry \
+		remove \
+		"$flavor" \
+		"$kernelrelease"-"$flavor" \
+		"$image"-"$kernelrelease"-"$flavor" \
+		initrd-"$kernelrelease"-"$flavor"
+	fi
+    fi
+
+    [ -z "$certs" ] || [ "$basepackage" -ne 1 ] || /usr/lib/suse-module-tools/kernel-cert-$op --certs "$certs" "$@"
+    ;;
+posttrans)
```

Comment:
While we are at it, please let's add a call to `/usr/lib/module-init-tools/regenerate-initrd-posttrans` here. It won't do anything anyway unless previous scriptlets left markers under `/run/regenerate-initrd`. Actually I would wish to have this call added to the kernel scriptlets before moving them to **s-m-t**.

Reply:
That's kind of pointless because we are just removing the scriptlets anyway.

There is no posttrans for the kernel but I added it here for completeness and for potentially moving the initrd invocation here in a separate bug/feature.

Reply:
Well, once it's in **s-m-t** I'll have it under my control :grin: